.github/workflows/node-ci-push.yml

```diff
@@ -35,36 +36,29 @@ jobs:
         with:
           node-version: ${{ matrix.node-version }}
 
-
-  coverage:
-    needs: [prime_cache_primary]
-    name: Gather coverage
-    env:
-      CI: true
+  lint:
+    name: Lint
+    needs: prime_cache_primary
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        # Use a matrix as it means we get the version info in the job name
-        # which is very helpful.
         os: [ubuntu-latest]
         node-version: [20.x]
     steps:
-    - uses: actions/checkout@v4
-    - name: Use Node.js ${{ matrix.node-version }}
-      uses: actions/setup-node@v4
-      with:
-        node-version: ${{ matrix.node-version }}
-
-    # Cache and install dependencies
-    - name: Install & cache node_modules
-      uses: Khan/actions@shared-node-cache-v2
-      with:
-        node-version: ${{ matrix.node-version }}
+      - uses: actions/checkout@v4
+      - uses: ./.github/actions/lint
 
-    - name: Run Jest with coverage
-      run: yarn coverage:ci
-    - name: Upload Coverage
-      uses: codecov/codecov-action@v4
-      with:
-        token: ${{ secrets.CODECOV_TOKEN }}
-        files: ./coverage/coverage-final.json
+  test:
+    name: Test
+    needs: prime_cache_primary
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        node-version: [20.x]
+        shard: ["1/2", "2/2"]
+    steps:
+      - uses: actions/checkout@v4
+      - uses: ./.github/actions/test
+        env:
+          CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
```

Comment:
One other difference between composite actions and reusing workflows is that [secrets can only be accessed by actions using an input or environment variable](https://docs.github.com/en/actions/security-for-github-actions/security-guides/using-secrets-in-github-actions#accessing-your-secrets), while [reusable workflows can define secrets](https://docs.github.com/en/actions/sharing-automations/reusing-workflows#using-inputs-and-secrets-in-a-reusable-workflow) 

Here, we pass the codecov token using an env variable. Let me know if there are any concerns with this!